clientv3/lease.go

```diff
@@ -244,6 +244,11 @@ func (l *lessor) KeepAlive(ctx context.Context, id LeaseID) (<-chan *LeaseKeepAl
 		l.mu.Unlock()
 		close(ch)
 		return ch, ErrKeepAliveHalted{Reason: err}
+	case <-ctx.Done():
```

Comment:
this is just for early termination on ctx.done right? I think old code is still correct.

Reply:
Yes, this is for early termination. Old code won't return context error for early context cancel/timeout.

Reply:
hmm, in the old code. what happen if ctx is cancelled early?

Reply:
No error is returned--only the keepalive channel is closed.

Reply:
ahh, will this breaks api in someway? since this code changes the behavior of api?